spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/reload/KubernetesClientEventBasedConfigMapChangeDetector.java

```diff
@@ -66,42 +96,46 @@ public KubernetesClientEventBasedConfigMapChangeDetector(CoreV1Api coreV1Api, Co
 		// certificate authorities for the cluster. This results in SSL errors.
 		// See https://github.com/spring-cloud/spring-cloud-kubernetes/issues/885
 		this.factory = new SharedInformerFactory(createApiClientForInformerClient());
-		this.namespace = kubernetesNamespaceProvider.getNamespace();
 		this.monitorConfigMaps = properties.isMonitoringConfigMaps();
+		this.enableReloadFiltering = properties.isEnableReloadFiltering();
+		namespaces = namespaces(kubernetesNamespaceProvider, properties, "configmap");
 	}
 
 	@PostConstruct
-	public void watch() {
-		if (coreV1Api != null && monitorConfigMaps) {
-			SharedIndexInformer<V1ConfigMap> configMapInformer = factory.sharedIndexInformerFor(
-					(CallGeneratorParams params) -> coreV1Api.listNamespacedConfigMapCall(namespace, null, null, null,
-							null, null, null, params.resourceVersion, null, params.timeoutSeconds, params.watch, null),
-					V1ConfigMap.class, V1ConfigMapList.class);
-			configMapInformer.addEventHandler(new ResourceEventHandler<>() {
-				@Override
-				public void onAdd(V1ConfigMap obj) {
-					log.info("ConfigMap " + obj.getMetadata().getName() + " was added.");
-					onEvent(obj);
-				}
+	void inform() {
+		if (monitorConfigMaps) {
+			log.info("Kubernetes event-based configMap change detector activated");
 
-				@Override
-				public void onUpdate(V1ConfigMap oldObj, V1ConfigMap newObj) {
-					log.info("ConfigMap " + newObj.getMetadata().getName() + " was added.");
-					onEvent(newObj);
-				}
+			namespaces.forEach(namespace -> {
+				SharedIndexInformer<V1ConfigMap> informer;
+				String filter = null;
 
-				@Override
-				public void onDelete(V1ConfigMap obj, boolean deletedFinalStateUnknown) {
-					log.info("ConfigMap " + obj.getMetadata() + " was deleted.");
-					onEvent(obj);
+				if (enableReloadFiltering) {
```

Comment:
if label filtering is enabled, create proper label-filter. There is an integration test in each client specific to this functionality